101-Linux-Java-ZuluOpenJDK/azuredeploy.json

```diff
@@ -0,0 +1,449 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "OS": {
+      "type": "string",
+      "defaultValue": "Ubuntu 18.04-LTS",
+      "allowedValues": [
+        "Ubuntu 18.04-LTS",
+        "Debian 9",
+        "Debian 10",
+        "CentOS 8.1"
+      ],
+      "metadata": {
+        "description": "The Linux distribution for the VM. This will pick a fully patched image of the selected distribution."
+      }
+    },
+    "javaPackageName": {
+      "type": "string",
+      "defaultValue": "Azul_Zulu_OpenJDK-8-JDK",
+      "allowedValues": [
+        "Azul_Zulu_OpenJDK-7-JDK",
+        "Azul_Zulu_OpenJDK-7-JRE",
+        "Azul_Zulu_OpenJDK-7-JRE-Headless",
+        "Azul_Zulu_OpenJDK-8-JDK",
+        "Azul_Zulu_OpenJDK-8-JRE",
+        "Azul_Zulu_OpenJDK-8-JRE-Headless",
+        "Azul_Zulu_OpenJDK-11-JDK",
+        "Azul_Zulu_OpenJDK-11-JRE",
+        "Azul_Zulu_OpenJDK-11-JRE-Headless",
+        "Azul_Zulu_OpenJDK-13-JDK",
+        "Azul_Zulu_OpenJDK-13-JRE",
+        "Azul_Zulu_OpenJDK-13-JRE-Headless"
+      ],
+      "metadata": {
+        "description": "Azul Zulu OpenJDK JVM for Azure package name"
+      }
+    },
+    "vmSize": {
+      "type": "string",
+      "defaultValue": "Standard_D2s_v3",
+      "metadata": {
+        "description": "Size for the Virtual Machine."
+      }
+    },
+    "vmName": {
+      "type": "string",
+      "defaultValue": "linux-zulu",
+      "metadata": {
+        "description": "Name for the Virtual Machine."
+      }
+    },
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "User name for the Virtual Machine."
+      }
+    },
+    "authenticationType": {
+      "type": "string",
+      "defaultValue": "sshPublicKey",
+      "allowedValues": [
+        "password",
+        "sshPublicKey"
+      ],
+      "metadata": {
+        "description": "Type of authentication to use on the Virtual Machine."
+      }
+    },
+    "adminPasswordOrKey": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Password or SSH key for the Virtual Machine."
+      }
+    },
+    "storageNewOrExisting": {
+      "type": "string",
+      "defaultValue": "new",
+      "metadata": {
+        "description": "Determines whether or not a new storage account should be provisioned."
+      }
+    },
+    "storageAccountName": {
+      "type": "string",
+      "defaultValue": "[concat('storage', uniqueString(resourceGroup().id))]",
+      "metadata": {
+        "description": "Name of the storage account"
+      }
+    },
+    "storageAccountType": {
```

Comment:
need to use allowedValues here since this is used for boot diags

Reply:
Is there a list of valid storageAccountTypes anywhere?

Reply:
I don't know - I know that premium is not support and suspect if you use the correct type, it will tell you what's supported.